### PR TITLE
feat(dev): Upgrade kafka and zookeeper to support Apple M1

### LINKED ISSUE #28570: Zookeper Docker image not working on Apple's arm64
We currently run 5.1.2 which is a couple of years old and does not run well on Apple M1's.

We could upgrade kafka and zookeeper to 6.2.0 where qemu panics don't happen.

We could also try not using zookeeper at all. See:
https://www.confluent.io/blog/kafka-without-zookeeper-a-sneak-peek/
https://github.com/getsentry/sentry/pull/28568

### PR DESCRIPTION
**Pull request by armenzg**
This version of zookeper does not throw qemu panics on Apple's M1.

See issue: https://github.com/confluentinc/kafka-images/issues/80#issuecomment-855511438

### CODE REVIEW

File: .github/actions/setup-sentry/action.yml [hunk 1]
```diff
@@ -162,7 +162,7 @@ runs:
             --name sentry_zookeeper \
             -d --network host \
             -e ZOOKEEPER_CLIENT_PORT=2181 \
-            confluentinc/cp-zookeeper:4.1.0
```
Comment (armenzg) [inline .github/actions/setup-sentry/action.yml:165]: It seems the CI was running an older version.

Comment (billyvg) [reply .github/actions/setup-sentry/action.yml:165]: What version do we run in production?

**Comment by armenzg** [comment]
I tried getting rid of zookeeper but it got complicated.

File: .github/actions/setup-sentry/action.yml (hunk 1 above)
Comment (armenzg) [reply .github/actions/setup-sentry/action.yml:165]: I believe we run [4.1.2-3](https://github.com/getsentry/ops/blob/c823e62f930ecc6c97bb08898c71e49edc7232f6/cookbooks/getsentry/attributes/default.rb#L643) and version [3.4.11](https://github.com/getsentry/ops/blob/c823e62f930ecc6c97bb08898c71e49edc7232f6/cookbooks/getsentry/attributes/default.rb#L631) for zookeeper.

Should I make the CI test against those versions?

As curiosity, we run a range of versions across the org: https://github.com/search?q=org%3Agetsentry+cp-kafka&type=code

Comment (billyvg) [reply .github/actions/setup-sentry/action.yml:165]: Yeah I think we should have CI match production, let's leave comments here

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -1724,15 +1724,15 @@ def build_cdc_postgres_init_db_volume(settings):
     ),
     "zookeeper": lambda settings, options: (
         {
-            "image": "confluentinc/cp-zookeeper:5.1.2",
+            "image": "confluentinc/cp-zookeeper:6.2.0",
```
Comment (billyvg) [inline src/sentry/conf/server.py:1727]: Can we leave comments here about the versions and why it differs from CI/prod

**Review by billyvg — COMMENTED** [review]
We should also change these in `getsentry`

File: .github/actions/setup-sentry/action.yml (hunk 1 above)
Comment (armenzg) [reply .github/actions/setup-sentry/action.yml:165]: @billyvg I've discovered that version of zookeeper we use in production is no longer available on Docker hub, thus, we cannot match the versions between CI and production.

FYI, this part of the PR is completely unnecessary to me.

**Comment by armenzg** [comment]
> We should also change these in `getsentry`

There's no changes required on `getsentry` as far as I know.

File: .github/actions/setup-sentry/action.yml (hunk 1 above)
Comment (armenzg) [reply .github/actions/setup-sentry/action.yml:165]: Asked about the versions picked in https://github.com/getsentry/sentry/pull/20543#pullrequestreview-755065322

File: src/sentry/conf/server.py (hunk 2 above)
Comment (armenzg) [reply src/sentry/conf/server.py:1731]: Updated.

**Review by billyvg — APPROVED** [review]

**Comment by avgupta456** [comment]
When developing locally with Docker today, I ran into a new issue of sentry_kafka and sentry_zookeeper docker containers continually crashing and restarting. The error logs match https://github.com/confluentinc/cp-docker-images/issues/883. I believe this is due to updating to 6.2.0 from 5.1.2 in conf/server.py. Reverting these two lines solves the issue locally.

**Comment by armenzg** [comment]
I created #28672 in case someone wants to revert this change while I'm off today.

**Comment by armenzg** [comment]
There's a fix in #28724

**Pull request merged**